libs/designsystem/shared/floating/src/floating.directive.spec.ts

```diff
@@ -0,0 +1,463 @@
+import { Component, ElementRef, Input, ViewChild } from '@angular/core';
+import { createComponentFactory, mockProvider, Spectator } from '@ngneat/spectator';
+import { PortalDirective } from '@kirbydesign/designsystem/shared/portal';
+import { TriggerEvent } from '@kirbydesign/designsystem/shared/floating';
+import { FloatingDirective } from './floating.directive';
+
+import * as floatingUi from '@floating-ui/dom';
+import { Placement } from '@floating-ui/core/src/types';
+import { Strategy } from '@floating-ui/dom';
+import any = jasmine.any;
+import { DesignTokenHelper } from '@kirbydesign/core';
+
+@Component({
+  template: `
+    <div #floatingElement></div>
+    <div
+      #hostElement
+      kirbyFloating
+      [reference]="floatingElementRef"
+      [isDisabled]="isDisabled"
+      [closeOnEscapeKey]="closeOnEscapeKey"
+      [closeOnSelect]="closeOnSelect"
+      [closeOnBackdrop]="closeOnBackdrop"
+      [strategy]="strategy"
+      [placement]="placement"
+    ></div>
+    <div #clickTarget></div>
+  `,
+  imports: [FloatingDirective],
+  standalone: true,
+})
+class FloatingTestComponent {
+  @ViewChild('floatingElement', { static: true }) public floatingElementRef: ElementRef;
+  @ViewChild('hostElement', { static: true }) public hostElementRef: ElementRef;
+  @ViewChild('clickTarget', { static: true }) public clickTargetRef: ElementRef;
+  @ViewChild(FloatingDirective, { static: true }) public floatingDirective: FloatingDirective;
+
+  /** Start values set to match directive */
+  @Input() public isDisabled: boolean = false;
+  @Input() public closeOnEscapeKey: boolean = true;
+  @Input() public closeOnSelect: boolean = true;
+  @Input() public closeOnBackdrop: boolean = true;
+  @Input() public strategy: Strategy = 'absolute';
+  @Input() public placement: Placement = 'bottom-start';
+}
+
+describe('FloatingDirective', () => {
+  let spectator: Spectator<FloatingTestComponent>;
+  let component: FloatingTestComponent;
+  let directive: FloatingDirective;
+
+  const createComponent = createComponentFactory({
+    component: FloatingTestComponent,
+    providers: [mockProvider(PortalDirective, {})],
+    detectChanges: false,
+  });
+
+  beforeEach(() => {
+    spectator = createComponent();
+    component = spectator.component;
+    directive = component.floatingDirective;
+  });
+
+  it('should be created', () => {
+    expect(spectator.component).toBeTruthy();
+  });
+
+  describe('input', () => {
+    describe('reference', () => {
+      it('should set', () => {
+        directive.reference = component.floatingElementRef;
+        expect(directive.reference).toEqual(component.floatingElementRef);
+      });
```

Comment:
This test seems redundant, as we forcibly assign the value on the first line, before checking what we just did. 

What do we actually want to test here?

Reply:
Unless there is a really good reason, I would not test the getter/setter functionality of Angular. The code for setting the local variable inside the component is very simple, and it is hard to test anything meaningful. 

This goes for all the 'should set' tests in my opinion.

Reply:
Generally I agree, I just thought for a component library, it seemed reasonable to be sure. 

Especially for reference, since the code is more than a simply setter:

`@Input() public set reference(ref: ElementRef) {
    this.tearDownEventHandling();
    this._reference = ref as ElementRef;
    this.setupEventHandling();
    this.autoUpdatePosition();
  }`
  
  So I'm making sure that if someone changes the affected code, we still get the right reference set.

Reply:
Okay, fair point!